packages/opal-client/opal_client/client.py

```diff
@@ -230,6 +246,17 @@ def _init_engine_runner(
                 piped_logs_format=opal_client_config.INLINE_CEDAR_LOG_FORMAT,
             )
 
+        # OpenFGA Runner 
```

Comment:
I appreciate the architectural suggestions, I believe such fundamental changes would be beyond the scope of this PR. 

The current implementation follows the established patterns used for both Cedar and OPA clients in the codebase. 

If a different architecture for policy engine clients was needed, it would warrant its own dedicated refactoring effort. For this PR, I focused on maintaining consistency with the current codebase while adding OpenFGA support.

Perhaps we could track these architectural suggestions as a separate enhancement issue for future consideration?

Reply:
@iwphonedo Thank you for the good recommendation, by now OPAL do have quite a few integrations, and the ergonomics of using / implementing either is already getting messy.

On the same note, like @daveads said, that really is out-of-scope for the context of this PR. I'll keep it in note as a potential suggestion to improve the reusability and extensibility of that code.

Noted 🙂

Reply:
fully understood